subprojects/diagnostics/src/main/java/org/gradle/api/reporting/dependents/internal/DependentComponentsRenderableDependency.java

```diff
@@ -54,6 +54,12 @@ public static DependentComponentsRenderableDependency of(ComponentSpec component
                 public boolean apply(BinarySpec binarySpec) {
                     return binarySpec.isBuildable();
                 }
+
+                @Override
+                // Added for Java 6 source compatibility
+                public boolean test(BinarySpec input) {
```

Comment:
If one forgets adding the override, how does the build fail? Is there a problem loading the class on Java 6/7? Or does compilation fail since the `default` method is not used when `target` is `1.6`?

Reply:
Compilation fails.